soc/arm/st_stm32/stm32h7/Kconfig.series

```diff
@@ -14,6 +14,7 @@ config SOC_SERIES_STM32H7X
 	select CLOCK_CONTROL_STM32_CUBE if CLOCK_CONTROL
 	select NEWLIB_LIBC
 	select USE_STM32_HAL_RCC_EX if CPU_CORTEX_M4
+	select HAS_SWO
```

Comment:
Please could you  crosscheck that SWO backend  you want to enable, is compatible with all the stm32 SoC?  
Regarding the SWO backend implementation, the expected path is ITM->TPIU->TRACESWO.
I took a look to the STM32F4 datasheet, it is ok. But regarding  stm32H7 and stm32MP1, the backend has changed, probably due to multi processor architecture...
At least for these two SoCs the TPIU no longer seems used but replaced by the SWO block to output on TRACESWO.
So not compatible with the current SWO backend

Reply:
You are right. Looking at the debug infrastructure block diagram for the STM32H742 (figure 816 in RM0433) and comparing it to the one for the STM32F437 (figure 485 in RM0090) shows that the TPIU is no longer routed to TRACESWO. I missed that. As you said, this works for the STM32F4 - that's the SoC I've used for testing - but it won't for some others in this patch. I'll re-examine and update the patch. Thanks for spotting this.

Reply:
I double-checked the reference manuals and reverted the changes related to the STM32H7, STM32MP1 and STM32WB SoCs, as they do not support SWO via the TPIU.